sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JSONOptions.scala

```diff
@@ -59,6 +62,8 @@ object JSONOptions {
     allowNumericLeadingZeros =
       parameters.get("allowNumericLeadingZeros").map(_.toBoolean).getOrElse(false),
     allowNonNumericNumbers =
-      parameters.get("allowNonNumericNumbers").map(_.toBoolean).getOrElse(true)
+      parameters.get("allowNonNumericNumbers").map(_.toBoolean).getOrElse(true),
```

Comment:
I'm sorry for lack of english. I know you are very busy
My intention is
- I agree it's good to allowNonNumericNumbers=true
- But There are "allowNonNumericNumbers=false" in comment or declaring
https://github.com/apache/spark/blob/master/python/pyspark/sql/readwriter.py#L161
https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala#L259
https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JSONOptions.scala#L34
- So I wondered if you were trying to set to false